clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/IngestionThrottler.java

```diff
@@ -105,18 +109,40 @@ public IngestionThrottler(
             false,
             EventThrottler.BLOCK_STRATEGY));
     if (serverConfig.isAdaptiveThrottlerEnabled()) {
-      VeniceAdaptiveIngestionThrottler adaptiveIngestionThrottler = new VeniceAdaptiveIngestionThrottler(
-          serverConfig.getNonCurrentVersionAAWCLeaderQuotaRecordsPerSecond(),
-          serverConfig.getKafkaFetchQuotaTimeWindow(),
-          "non_current_version_aa_wc_leader_records_count");
-      this.poolTypeRecordThrottlerMap
-          .put(ConsumerPoolType.NON_CURRENT_VERSION_AA_WC_LEADER_POOL, adaptiveIngestionThrottler);
-      adaptiveIngestionThrottler = new VeniceAdaptiveIngestionThrottler(
-          serverConfig.getNonCurrentVersionNonAAWCLeaderQuotaRecordsPerSecond(),
-          serverConfig.getKafkaFetchQuotaTimeWindow(),
-          "non_current_version_non_aa_wc_leader_records_count");
-      this.poolTypeRecordThrottlerMap
-          .put(ConsumerPoolType.NON_CURRENT_VERSION_NON_AA_WC_LEADER_POOL, adaptiveIngestionThrottler);
+      VeniceAdaptiveIngestionThrottler nonCurrentVersionAaWcLeaderAdaptiveIngestionThrottler =
+          new VeniceAdaptiveIngestionThrottler(
+              serverConfig.getNonCurrentVersionAAWCLeaderQuotaRecordsPerSecond(),
+              serverConfig.getKafkaFetchQuotaTimeWindow(),
+              "non_current_version_aa_wc_leader_records_count");
+      nonCurrentVersionAaWcLeaderAdaptiveIngestionThrottler
+          .registerLimiterSignal(adaptiveThrottlerSignalService::isSingleGetLatencySignalActive);
+      nonCurrentVersionAaWcLeaderAdaptiveIngestionThrottler
+          .registerLimiterSignal(adaptiveThrottlerSignalService::isCurrentLeaderMaxHeartbeatLagSignalActive);
+      nonCurrentVersionAaWcLeaderAdaptiveIngestionThrottler
+          .registerLimiterSignal(adaptiveThrottlerSignalService::isCurrentFollowerMaxHeartbeatLagSignalActive);
+      nonCurrentVersionAaWcLeaderAdaptiveIngestionThrottler
+          .registerBoosterSignal(adaptiveThrottlerSignalService::isNonCurrentLeaderMaxHeartbeatLagSignalActive);
```

Comment:
why is non-current lag is a booster signal ?

Reply:
My thinking is, current lag is high priority so it is the limiter for the non-current adaptive throttler. If they are good, then we can check if we can bump up the ingestion for non-current as well. The boost signal then should be coming from non-current lag